Location Rule Cleanup

Seatori opened the pull request:
Hidden in the Depths is triggered by having Ancient Debris in your inventory, not Netherite Scrap.
Cover Me in Debris and Serious Dedication currently have incorrect logic rules that don't align perfectly with their physical access. This accounts for that so that they're only shown as available out of logic when their physical access conditions are met, and only shown in logic when their logical access conditions are met.

Tested by toggling various options in the PopTracker and verifying that they matched up with the correct access/logic rules.

Seatori committed:
Location Rule Cleanup

Hidden in the Depths is triggered by having Ancient Debris in your inventory, not Netherite Scrap.
Cover Me in Debris and Serious Dedication currently have incorrect logic rules that don't align with their physical access. This accounts for that so that they're only shown out of logic when their physical access conditions are met, and only shown in logic when their logical access conditions are met.

Seatori committed:
Update Hidden in the Depths

Diamond pickaxes can be gotten by trading with cured villagers.

Cyb3RGER commented:
I just released a new version with palex fixes 😅 
Scraps not unlocking Hidden in the Depths makes sense to me but the other two would now be possible out of logic with nothing. that doesn't seem right.
Also Hidden in the Depths surely needs diamond pickaxe to be in logic normally, right?

Seatori commented:
> I just released a new version with palex fixes 😅 Scraps not unlocking Hidden in the Depths makes sense to me but the other two would now be possible out of logic with nothing. that doesn't seem right. Also Hidden in the Depths surely needs diamond pickaxe to be in logic normally, right?

I'm patching in some fixes now. Wrote this hastily when I saw the other one was merged so I got the requirements backwards.

Cyb3RGER commented:
Ok put this in draft then for now, just convert back when you are done with your changes.
Would also be nice if you can test them as well.

Seatori commented:
Will do.
I'm testing them as I go.

Seatori committed:
Fix Cover Me in Debris and Serious Dedication

They no longer show as being always available out of logic, and instead are correctly only out of logic when a Bastion can be accessed and only in logic once their proper rules have been fulfilled.
